### PR TITLE
Simplify user interface, use ros standard srvs

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Mixes 2 things. If not acceptable, can remove either one of the changes.

If needed, can close the PR :)

File: depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp [hunk 1]
```diff
@@ -247,27 +249,28 @@ class DeviceROS : public dai::DeviceBase {
...
                     _pub_t[name] =
-                            std::thread{generate_pub_lambda<depthai_datatype_msgs::RawImgFrame>(_pub_nh, name, 10)};
+                            std::thread{generate_pub_lambda<depthai_datatype_msgs::RawImgDetections>(nh, name, 10)};
                     break;
+                case dai::DatatypeEnum::ImgFrame: {
+                    _pub_t[name] = std::thread{generate_pub_lambda<depthai_datatype_msgs::RawImgFrame>(nh, name, 10)};
+                    boost::function<bool(std_srvs::Trigger::Request&, std_srvs::Trigger::Response&)> fn =
+                            boost::bind(&DeviceROS::_defaultCameraInfo, this, name, ph::_1, ph::_2);
+                    _camera_info_default_srvs[name] = nh.advertiseService("reset_camera_info", fn);
+                } break;
```
Comment (kota-rr) [inline depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:271]: line change for break?

File: depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp [hunk 2]
```diff
@@ -247,27 +249,28 @@ class DeviceROS : public dai::DeviceBase {
...
-                case dai::DatatypeEnum::ImgFrame:
                     _pub_t[name] =
-                            std::thread{generate_pub_lambda<depthai_datatype_msgs::RawImgFrame>(_pub_nh, name, 10)};
+                            std::thread{generate_pub_lambda<depthai_datatype_msgs::RawImgDetections>(nh, name, 10)};
                     break;
+                case dai::DatatypeEnum::ImgFrame: {
+                    _pub_t[name] = std::thread{generate_pub_lambda<depthai_datatype_msgs::RawImgFrame>(nh, name, 10)};
+                    boost::function<bool(std_srvs::Trigger::Request&, std_srvs::Trigger::Response&)> fn =
+                            boost::bind(&DeviceROS::_defaultCameraInfo, this, name, ph::_1, ph::_2);
+                    _camera_info_default_srvs[name] = nh.advertiseService("reset_camera_info", fn);
```
Comment (kota-rr) [inline depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:270]: Is this service for resetting RoI control?

**Review by kota-rr — APPROVED** [review]

File: depthai_ros_driver/include/depthai_ros_driver/conversion.hpp [hunk 3]
```diff
@@ -129,17 +129,16 @@ struct adapt_dai2ros<depthai_datatype_msgs::RawImgFrame> {
 
     static ImagePublishers create_publisher(ros::NodeHandle& nh, const std::string& name, std::size_t q_size) {
         ImagePublishers pubs;
-        pubs.raw_image_pub = nh.advertise<OutputType>(name + "/image_raw", q_size);
-        pubs.compressed_image_pub = nh.advertise<OutputType>(name + "/image_raw/compressed", q_size);
-        pubs.camera_info_pub = nh.advertise<sensor_msgs::CameraInfo>(name + "/camera_info", q_size);
+        pubs.raw_image_pub = nh.advertise<OutputType>("image_raw", q_size);
+        pubs.compressed_image_pub = nh.advertise<OutputType>("image_raw/compressed", q_size);
+        pubs.camera_info_pub = nh.advertise<sensor_msgs::CameraInfo>("camera_info", q_size);
```
Comment (ardiya) [inline depthai_ros_driver/include/depthai_ros_driver/conversion.hpp:134]: I still don't get why are we removing the name from the publisher?

**Review by ardiya — APPROVED** [review]
Looks Good to Me

File: depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp (hunk 2 above)
Comment (kunaltyagi) [reply depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:270]: No, this is the service for setting the camera info to default

File: depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp (hunk 1 above)
Comment (kunaltyagi) [reply depthai_ros_driver/include/depthai_ros_driver/device_ros.hpp:271]: Not needed. This is auto-format based on the clang-format file supplied in the project.

**Pull request merged**